app/src/main/java/org/oppia/android/app/home/HomeFragmentPresenter.kt

```diff
@@ -96,9 +102,49 @@ class HomeFragmentPresenter @Inject constructor(
       it.viewModel = homeViewModel
     }
 
+    logAppOnboardedEvent()
+
     return binding.root
   }
 
+  private fun logAppOnboardedEvent() {
+    val startupStateProvider = appStartupStateController.getAppStartupState()
+    val liveData = startupStateProvider.toLiveData()
+    liveData.observe(
+      activity,
+      object : Observer<AsyncResult<AppStartupState>> {
+        override fun onChanged(startUpStateResult: AsyncResult<AppStartupState>?) {
+          when (startUpStateResult) {
+            is AsyncResult.Pending -> {
+              // Do nothing
+            }
+            is AsyncResult.Success -> {
+              // It's possible for the observer to still be active & change due to the next activity
+              // causing a notification to be posted. That's always invalid to process here: the
+              // splash activity should never do anything after its initial state since it always
+              // finishes (or in the case of the deprecation dialog, blocks) the activity.
+              liveData.removeObserver(this)
+
+              if (startUpStateResult.value.startupMode ==
+                AppStartupState.StartupMode.USER_NOT_YET_ONBOARDED
```

Comment:
Once the user launches the app for the first time, the StartupMode will still be USER_NOT_YET_ONBOARDED. This will change (forever) on the next launch so it is safe enough to use it this way.